testnet/stacks-node/src/event_dispatcher.rs

```diff
@@ -573,11 +602,14 @@ impl EventObserver {
         })
     }
 
-    /// Returns tuple of (txid, success, raw_result, raw_tx, contract_interface_json)
-    fn generate_payload_info_for_receipt(receipt: &StacksTransactionReceipt) -> ReceiptPayloadInfo {
+    /// Returns transaction event payload to send for new block or microblock event
+    fn make_new_block_txs_payload(
```

Comment:
Could you add a unit test for this function, ensuring that there is a transaction with and a transaction without a `vm_error`?